Steer/DigitizerWorkflow/src/CTPDigitizerSpec.cxx

```diff
@@ -0,0 +1,105 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "CTPDigitizerSpec.h"
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework//Task.h"
+#include "DetectorsBase/BaseDPLDigitizer.h"
+#include "DataFormatsCTP/Digits.h"
+#include "Steer/HitProcessingManager.h" // for DigitizationContext
+#include "DetectorsCommonDataFormats/DetID.h"
+#include "CTPSimulation/Digitizer.h"
+#include "DataFormatsFT0/Digit.h"
+#include "DataFormatsFV0/BCData.h"
+
+#include <TStopwatch.h>
+#include <gsl/span>
+
+using namespace o2::framework;
+namespace o2
+{
+namespace ctp
+{
+class CTPDPLDigitizerTask : public o2::base::BaseDPLDigitizer
+{
+  using GRP = o2::parameters::GRPObject;
+
+ public:
+  CTPDPLDigitizerTask() : o2::base::BaseDPLDigitizer(), mDigitizer() {}
+  ~CTPDPLDigitizerTask() override = default;
+  void initDigitizerTask(framework::InitContext& ic) override
+  {
+    mDigitizer.init();
+  }
+  void run(framework::ProcessingContext& pc)
+  {
+    // read collision context from input
+    //auto context = pc.inputs().get<o2::steer::DigitizationContext*>("collisioncontext");
+    //const bool withQED = context->isQEDProvided();
+    //auto& timesview = context->getEventRecords(withQED);
+    // read ctp inputs from input
+    auto ft0inputs = pc.inputs().get<gsl::span<o2::ft0::DetTrigInput>>("ft0");
+    auto fv0inputs = pc.inputs().get<gsl::span<o2::fv0::DetTrigInput>>("fv0");
+
+    // if there is nothing to do ... return
+    if ((ft0inputs.size() == 0) && (fv0inputs.size() == 0)) {
+      return;
+    }
+    std::vector<o2::ctp::CTPInputDigit> finputs;
+    TStopwatch timer;
+    timer.Start();
+    LOG(INFO) << "CALLING CTP DIGITIZATION";
+    for (const auto& inp : ft0inputs) {
+      CTPInputDigit finpdigit(inp.mIntRecord, inp.mInputs, o2::detectors::DetID::FT0);
+      finputs.emplace_back(finpdigit);
+    }
+    for (const auto& inp : fv0inputs) {
+      CTPInputDigit finpdigit(inp.mIntRecord, inp.mInputs, o2::detectors::DetID::FT0);
+      finputs.emplace_back(finpdigit);
+    }
+    gsl::span<CTPInputDigit> ginputs(finputs);
+    mDigitizer.process(ginputs, mDigits);
```

Comment:
this will not work, you actually don't need this `mDigits`: if you accept my suggestion above on returning the vector from Digitize, then you can do directly:
````
auto digits = mDigitizer.process(ginputs);
pc.outputs().snapshot(Output{"CTP", "DIGITS", 0, Lifetime::Timeframe}, digits);
````

Reply:
yes, of course, all accepted.